bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/config/BigtableOptions.java

```diff
@@ -544,4 +469,12 @@ public Builder toBuilder() {
   public boolean useCachedChannel() {
     return useCachedDataPool;
   }
+
+  protected BigtableOptions clone() {
+    try {
+      return (BigtableOptions) super.clone();
```

Comment:
Wouldn't this inadvertently update the reference type fields already in use before the clone, and effectively make them mutable?

Reply:
You mean objects like `CallOptionsConfig` and `RetryOptions`?  Those are supposed to be immutable.